.github/workflows/macos.yml

```diff
@@ -77,39 +77,28 @@ jobs:
             wget --quiet https://gstreamer.freedesktop.org/data/pkg/osx/${{ env.GST_VERSION }}/gstreamer-1.0-devel-${{ env.GST_VERSION }}-universal.pkg
             wget --quiet https://gstreamer.freedesktop.org/data/pkg/osx/${{ env.GST_VERSION }}/gstreamer-1.0-${{ env.GST_VERSION }}-universal.pkg
             for package in *.pkg ;
-              do sudo installer -verbose -pkg "$package" -target /
+            do sudo installer -verbose -pkg "$package" -target /
             done
-            echo "PKG_CONFIG_PATH=/Library/Frameworks/GStreamer.framework/lib/pkgconfig:${{ env.PKG_CONFIG_PATH }}" >> "$GITHUB_ENV"
-
+  
+      - name: Install Dependencies
+        run:  |
+            brew install ninja
+    
       - name: Create build directory
         run:  mkdir ${{ runner.temp }}/shadow_build_dir
 
       - name: Configure
         working-directory: ${{ runner.temp }}/shadow_build_dir
-        run: |
-          if [ ${{ matrix.BuildType }} == Debug ]
-          then
-            cmake -S ${{ github.workspace }} -B . -G Ninja \
-              -DCMAKE_BUILD_TYPE=${{ matrix.BuildType }} \
+        run: cmake -S ${{ github.workspace }} -B . -G Ninja
+              -DCMAKE_BUILD_TYPE=${{ matrix.BuildType }}
+              -DQGC_BUILD_TESTING=${{ matrix.BuildType == 'Debug' && 'ON' || 'OFF' }}
```

Comment:
Since we only care to run testing on linux, i would omit -DQGC_BUILD_TESTING otherwise it will build it for debug runs

Reply:
In CI sure we only run unit tests on Linux. But if we are going to build debug anyway we might as well take advantage of doing a full build. If all someone has local dev is a specific OS, then their debug build on the specific Os is what they do unit tests on locally. Might as well validate that.

Reply:
Also I'm not sure what the defaults or now in cmake build. But Debug should just default unit tests compiled in and Release to not.